src/main/java/minicraft/saveload/Load.java

```diff
@@ -437,99 +456,198 @@ private void loadUnlocks(String filename) {
+				for (minicraft.gfx.Point p: curLevel.getMatchingTiles(Tiles.get("Stairs Up"))) {
+					if (parent.getTile(p.x, p.y) != Tiles.get("Stairs Down")) {
+						parent.printLevelLoc("INCONSISTENT STAIRS detected; placing stairsDown", p.x, p.y);
+						parent.setTile(p.x, p.y, Tiles.get("Stairs Down"));
+					}
 				}
 			}
-			for (minicraft.gfx.Point p: curLevel.getMatchingTiles(Tiles.get("Stairs Up"))) {
-				if (parent.getTile(p.x, p.y) != Tiles.get("Stairs Down")) {
-					parent.printLevelLoc("INCONSISTENT STAIRS detected; placing stairsDown", p.x, p.y);
-					parent.setTile(p.x, p.y, Tiles.get("Stairs Down"));
+		} else {
+			for(int l = World.maxLevelDepth; l >= World.minLevelDepth; l--) {
+				LoadingDisplay.setMessage(Level.getDepthString(l));
+				int lvlidx = World.lvlIdx(l);
+				JSONObject data = new JSONObject(loadFromFile(location + filename + lvlidx + extension, true));
+				
+				int lvlw = data.getInt("width");
+				int lvlh = data.getInt("height");
+	
+				long seed = data.getLong("seed");
+				Settings.set("size", lvlw);
+				
+				short[] tiles = new short[lvlw * lvlh];
+				JSONObject[] tdata = new JSONObject[lvlw*lvlh];
+				
+				JSONArray saveTiles = data.getJSONArray("tiles");
+				for (int x = 0; x < lvlw; x++) {
+					for (int y = 0; y < lvlh; y++) {
+						int tileArrIdx = y + x * lvlw;
+						int tileidx = x + y * lvlw; // the tiles are saved with x outer loop, and y inner loop, meaning that the list reads down, then right one, rather than right, then down one.
+						JSONObject tile = saveTiles.getJSONObject(tileidx);
+						tiles[tileArrIdx] = Tiles.get(tile.getString("name")).id;
+						tdata[tileArrIdx] = tile.getJSONObject("data");
+					}
+				}
+				
+				Level parent = World.levels[World.lvlIdx(l+1)];
+				World.levels[lvlidx] = new Level(lvlw, lvlh, seed, l, parent, false);
+				
+				Level curLevel = World.levels[lvlidx];
+				curLevel.tiles = tiles;
+				curLevel.data = tdata;
+				
+				if (Game.debug) curLevel.printTileLocs(Tiles.get("Stairs Down"));
+				
+				if (parent == null) continue;
+				/// confirm that there are stairs in all the places that should have stairs.
+				for (minicraft.gfx.Point p: parent.getMatchingTiles(Tiles.get("Stairs Down"))) {
+					if (curLevel.getTile(p.x, p.y) != Tiles.get("Stairs Up")) {
+						curLevel.printLevelLoc("INCONSISTENT STAIRS detected; placing stairsUp", p.x, p.y);
+						curLevel.setTile(p.x, p.y, Tiles.get("Stairs Up"));
+					}
+				}
+				for (minicraft.gfx.Point p: curLevel.getMatchingTiles(Tiles.get("Stairs Up"))) {
+					if (parent.getTile(p.x, p.y) != Tiles.get("Stairs Down")) {
+						parent.printLevelLoc("INCONSISTENT STAIRS detected; placing stairsDown", p.x, p.y);
+						parent.setTile(p.x, p.y, Tiles.get("Stairs Down"));
+					}
 				}
 			}
 		}
 	}
 	
-	public void loadPlayer(String filename, Player player) {
+	public void loadPlayer(String filename, Player player) throws JSONException, IOException {
```

Comment:
These should absolutely not throw exceptions. Errors should be handled where they occur, so you can write a proper error message.